Lib/test/test_urllibnet.py

```diff
@@ -213,14 +214,44 @@ def test_context_argument(self):
         self.assertIn("Python", response.read())
 
 
+class urlopen_FTPTest(unittest.TestCase):
+    # TODO: https://github.com/python/pythondotorg/issues/1069
+    # Use a resource hosted at python.org
+    FTP_TEST_FILE = 'ftp://ftp.debian.org/debian/README'
+
+    def test_multiple_ftp_retrieves(self):
+
+        with test_support.transient_internet(self.FTP_TEST_FILE):
+            try:
+                for _ in range(3):
+                    urllib.FancyURLopener().retrieve(
+                            self.FTP_TEST_FILE,
+                            tempfile.NamedTemporaryFile().name)
+
+            except IOError as e:
+                self.fail("Failed FTP retrieve while accessing ftp url "
+                          "multiple times.\n Error message was : %s" % e)
+
+    def test_multiple_ftp_urlopen_same_host(self):
+        with test_support.transient_internet(self.FTP_TEST_FILE):
+            try:
+                for _ in range(3):
+                    fp = urllib.urlopen(self.FTP_TEST_FILE)
+                    # closing  fp is not required.
```

Comment:
This test can behave differently in CPython and PyPy. If fp can remain unclosed, maybe save it in a list, so that it will not be closed in all implementations? If the file is not closed explicitly it is worth to call test_support.gc_collect() after removing all references to it. This will make the behavior more predicable in all implementations.